src/python/zigopt/protobuf/lib.py

```diff
@@ -1,54 +1,37 @@
 # Copyright © 2022 Intel Corporation
 #
 # SPDX-License-Identifier: Apache License 2.0
+from typing import Any, Generic, TypeVar
+
 from google.protobuf.message import Message
 
 from zigopt.common import *
 
 
-# sigoptlint: disable=ProtobufMethodsRule
-
-
-def CopyFrom(base, other):
-  # NOTE: Soon this will be replaced with a safer implementat, but add a stub and lint rule
-  # for now to make merging easier
-  return base.CopyFrom(other)
-
-
-def MergeFrom(base, other):
-  # NOTE: Soon this will be replaced with a safer implementat, but add a stub and lint rule
-  # for now to make merging easier
-  return base.MergeFrom(other)
-
+MessageT = TypeVar("MessageT", bound=Message)
 
-def GetFieldOrNone(underlying, name):
-  if name in underlying.DESCRIPTOR.oneofs_by_name:
-    raise ValueError(f"Cannot call GetFieldOrNone on oneof field: {name}")
-  if underlying.HasField(name):
-    return getattr(underlying, name)
-  return None
 
-
-def SetFieldIfNotNone(underlying, name, value):
-  # Ensure that we always raise on invalid attributes, even if value is None
-  if not hasattr(underlying, name):
-    raise AttributeError(name)
-
-  if value is not None:
-    setattr(underlying, name, value)
-
-
-def GetOneofValueOrNone(underlying, name):
-  which_oneof = underlying.WhichOneof(name)
+def get_oneof_value(proto: MessageT, name: str) -> Any:
+  which_oneof = proto.WhichOneof(name)
   if which_oneof is not None:
-    return getattr(underlying, which_oneof)
-  return None
+    return getattr(proto, which_oneof)
+  raise TypeError(f"No value set for oneof: {name}")
```

Comment:
We use this in 1 place. So in that place I just catch the `TypeError` and handle appropriately. Since we only do this in 1 place I could just inline the logic.